src/modules/rolle/specification/only-assignable-sps.ts

```diff
@@ -0,0 +1,32 @@
+import { xor } from 'lodash-es';
+import { ServiceProvider } from '../../service-provider/domain/service-provider.js';
+import { CompositeSpecification } from '../../specification/specifications.js';
+
+import { ServiceProviderMerkmal } from '../../service-provider/domain/service-provider.enum.js';
+import { ServiceProviderRepo } from '../../service-provider/repo/service-provider.repo.js';
+import { Rolle } from '../domain/rolle.js';
+export class NurNachtraeglichZuweisbareServiceProvider extends CompositeSpecification<Rolle<boolean>> {
+    public constructor(
+        private readonly serviceProviderRepo: ServiceProviderRepo,
+        private readonly oldRolle: Rolle<true>,
+    ) {
+        super();
+    }
+
+    public async isSatisfiedBy(updatedRolle: Rolle<boolean>): Promise<boolean> {
+        const changedServiceProviderIds: string[] = xor(
+            this.oldRolle.serviceProviderIds,
+            updatedRolle.serviceProviderIds,
+        );
```

Comment:
```suggestion
        const changedServiceProviderIds: string[] = xor(
            this.oldRolle.serviceProviderIds,
            updatedRolle.serviceProviderIds,
        );
        
          if (changedServiceProviderIds.length === 0) {
          // nothing changed, so the constraint is trivially satisfied
          return true;
           }
```
Early return if nothing was changed (Since its possible to make send a request through the UI with no change at all)